File: sorting/insertion/insertion.cpp
#include <iostream>
#include "readintarray.h"
#include "printintarray.h"

void insertionSort(int *array, int len){
	int i, j, key;
	
	for(j = 1; j < len; j++){
		i = j - 1;
		key = array[j];

		while(i >= 0 && array[i] > key){
			array[i + 1] = array[i];
			i--;
		}

		array[i + 1] = key;
	}
}

int main(){
	int int_list[BUFFSIZE];
	int len;

	readIntArray(int_list, &len);
	insertionSort(int_list, len);
	printIntArray(int_list, len);

	return 0;
}

File: sorting/insertion/insertion.py
#!/bin/python2

def insertion_sort( l ):
	for j in range(1, len(l)):
		i = j - 1
		key = l[j]

		while (i >= 0 and l[i] > key):
			l[i + 1] = l[i]
			i = i - 1

		l[i + 1] = key

inp = raw_input("Enter integer list.\n>>> ")
int_list = [int(x) for x in inp.split(',')]
insertion_sort(int_list)
print int_list
